module-2/backend/week_one.md

```diff
@@ -5,30 +5,78 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+  * PATCH
 2. What is Sinatra?
+  * It's a framework for building websites. A domain specific language
 4. What is MVC?
+  * It's a convention for writing web applications. Model, View, Controller
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  * So that other people can understand what we are doing. 
 6. What types of variables are accessible in our view templates without explicitly passing them?
+  * Instance variables
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
     erb :index
   end
+  
+  index.erb
+  <p><%= @count %></p>
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed`?
+  erb :index, :locals => {:number => @@number,
+    ```ruby
+  get '/horses' do
+    name = Mr. Ed
+    erb :index, :locals => {:name => name}
+  end
+  ```
 9. What's the purpose of ERB?
+  * So that we can render ruby code in an html form, so that it can be presented to a user using our web application.
 10. Why do I need a development AND test database?
+  * Our testing would mess up the development database by adding, updating, and deleting. We also need to have a fresh empty database each time we test. This is why we have the test database so we can test and keep out testing separate from the development.
 11. What's responsive design?
+  * It is when we design a website that can be responsive to different screens(e.g. mobile to desktop and vice versa) and still be usable.
 12. What is CRUD and why is it important?
+  * Create, Read, Update, and Delete. It allows a user to interact with a database giving them full access to the data.
 13. What does HTTP stand for? 
+  * Hyper text transfer protocol
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  * <%= %> (this displays ruby on the website), <% %> this allows the ruby to run without being displayed.
 15. What's an ORM?
+  * Object Relation Model. It allows to relate objects with a database.
 16. What's the most commonly used ORM?
+  * SQL
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+  * '/restaurants'
+    *Shows all restaurants
+  * '/restaurants/:id'
+    *Shows one restaurant
+  * '/restraunts/new
+    *Adds a new restaurant
+  * '/restaurants/:id/edit'
+    *Shows the user the form that pdates/edits the current information for the current restaurant
+  * '/restaurants/:id'
+    *Brings the user to the newly updated restaurant
+  * '/restaurants/:id'
+    *Allows the user to delete the restaurant
+  * '/restaurants'
+    *After the deletion, the user is redirected to the page that shows all restaurants.

```

Comment:
These URL patterns need to be paired with verbs to really get the whole picture. So, for example, you have `/restaurants/:id` a few times above, but that pattern does different things depending on the verb. So a GET to that path will show a specific restaurant, while a PUT will update that restaurant.